src/package.rs

```diff
@@ -41,23 +41,90 @@ use std::str::FromStr;
 /// let mut package = Package::new(vec![my_deck], vec!["sound.mp3", "images/image.jpg"])?;
 /// package.write_to_file("output.apkg")?;
 /// ```
+/// # Example (media files from memory):
+/// ```rust	
+/// use genanki_rs::{Package, Deck, Note, Model, Field, Template, MediaFile};
+
+/// const VALID_MP3: &[u8] =
+/// b"\xff\xe3\x18\xc4\x00\x00\x00\x03H\x00\x00\x00\x00LAME3.98.2\x00\x00\x00\
+/// \x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\
+/// \x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\
+/// \x00\x00\x00\x00\x00\x00\x00\x00\x00\x00";
+/// 
+/// const VALID_JPG: &[u8] =
+/// b"\xff\xd8\xff\xdb\x00C\x00\x03\x02\x02\x02\x02\x02\x03\x02\x02\x02\x03\x03\
+/// \x03\x03\x04\x06\x04\x04\x04\x04\x04\x08\x06\x06\x05\x06\t\x08\n\n\t\x08\t\
+/// \t\n\x0c\x0f\x0c\n\x0b\x0e\x0b\t\t\r\x11\r\x0e\x0f\x10\x10\x11\x10\n\x0c\
+/// \x12\x13\x12\x10\x13\x0f\x10\x10\x10\xff\xc9\x00\x0b\x08\x00\x01\x00\x01\
+/// \x01\x01\x11\x00\xff\xcc\x00\x06\x00\x10\x10\x05\xff\xda\x00\x08\x01\x01\
+/// \x00\x00?\x00\xd2\xcf \xff\xd9";
+/// 
+/// let model = Model::new(
+/// 1607392319,
+/// "Simple Model",
+/// vec![
+///     Field::new("Question"),
+///     Field::new("Answer"),
+///     Field::new("MyMedia"),
+/// ],
+/// vec![Template::new("Card 1")
+///     .qfmt("{{Question}}{{Question}}<br>{{MyMedia}}")
+///     .afmt(r#"{{FrontSide}}<hr id="answer">{{Answer}}"#)],
+/// );
+/// let mut deck = Deck::new(1234, "Example Deck", "Example Deck with media");
+/// deck.add_note(Note::new(model.clone(), vec!["What is the capital of France?", "Paris", "[sound:sound.mp3]"])?);
+/// deck.add_note(Note::new(model.clone(), vec!["What is the capital of France?", "Paris", r#"<img src="image.jpg">"#])?);
+/// 
+/// let mut package = Package::new_from_memory(vec![deck], vec![MediaFile::new_from_bytes(VALID_MP3, "sound.mp3"), MediaFile::new_from_bytes(VALID_JPG, "image.jpg")])?;
+/// package.write_to_file("output.apkg")?;
+/// ```
 pub struct Package {
     decks: Vec<Deck>,
-    media_files: Vec<PathBuf>,
+    media_files: Vec<MediaFile>,
 }
+/// the location of the media files, either as a path on the filesystem or as bytes from memory
+pub enum MediaFile {
+    /// a path on the filesystem
+    Path(PathBuf),
+    /// bytes of the file and a filename
+    Bytes(Vec<u8>, String),
+}
+impl MediaFile {
+    /// Create a new `MediaFile` from a path on the filesystem
+    pub fn new_from_file<P: AsRef<Path>>(path: P) -> Self {
+        Self::Path(path.as_ref().to_path_buf())
+    }
 
+    /// Create a new `MediaFile` from a path on the filesystem using a `&str`
+    pub fn new_from_file_path(path: &str) -> Result<Self, Error> {
+        Ok(Self::Path(PathBuf::from_str(path)?))
+    }
+
+    /// Create a new `MediaFile` from bytes from memory and a filename
+    pub fn new_from_bytes(bytes: &[u8], name: &str) -> Self {
+        Self::Bytes(bytes.to_vec(), name.to_owned())
```

Comment:
Would it be possible not to call `to_vec` here and just keep a reference to the bytes? Could potentially avoid a chunky memcopy